src/adapter/src/catalog/builtin.rs

```diff
@@ -3258,8 +3260,16 @@ pub const MZ_CLUSTER_REPLICA_HISTORY: BuiltinView = BuiltinView {
                 WHERE object_type = 'cluster-replica' AND event_type = 'drop'
             )
         SELECT
-            creates.replica_id,
+            -- [btv] We have two notions of replica ID: the old one,
+            -- which is only present in `mz_audit_events`, and
+            -- the new one. We convert from old to new by
+            -- prepending with `u` here,
+            -- so as to not leak implementation details of `mz_audit_events`
+            -- into this more-broadly-useful view.
+            'u' || creates.replica_id AS replica_id,
```

Comment:
@umanwizard and I discussed live that this view should just pass the `replica_id` straight through, and we'll solve the ID migration issue independently of this PR.